src/main/java/org/inferred/freebuilder/processor/Analyser.java

```diff
@@ -312,6 +310,17 @@ private static boolean isUnderride(ExecutableElement method) {
     return userClass;
   }
 
+  private Optional<BuilderFactory> getBuilderFactory(Optional<TypeElement> builder) {
```

Comment:
I would just call it builderFactory - I consider verbs optional, especially if the method implements a lazy or O(1) operation.